### PR TITLE
[docs](update) Release 3.0.0

### PR DESCRIPTION
**Pull request by KassieZ**

### CODE REVIEW

**Review by luzhijing — APPROVED** [review]
LGTM

File: versioned_docs/version-3.0/releasenotes/release-3.0.0.md [hunk 1]
```diff
@@ -4,3 +4,166 @@
...
+
+- All DQL and DML, except for load operations, will use the new optimizer. Default settings disable fallback to the old optimizer, which will be removed in subsequent versions.
+
+- Comprehensive top-N runtime filter applicable to all query modes, enhancing performance when a top-N operator is present.
+
+- Experimental Feature: Refactored logic for synchronized materialized view selection, transitioning from RBO to CBO to maintain consistency with asynchronous materialized views. If issues arise, the switch `set global enable_sync_mv_cost_based_rewrite = false` can be used to revert to RBO mode.
+
+- Experimental Feature: Partition-level statistics. By activating the switch `set global enable_partition_analyze = true`, partition-level statistics can be collected and utilized, improving query performance in cases of data partition skew.
+
+- Experimental Feature: Generated columns. When creating tables, columns can be designated as generated columns, which automatically compute results based on defined expressions at write time, offering more complex expressions compared to default values but disallowing explicit writing of specified values.
```
Comment (johnpyp) [inline versioned_docs/version-3.0/releasenotes/release-3.0.0.md:113]: Suggestion: Link to the corresponding docs for Generated Columns

File: versioned_docs/version-3.0/releasenotes/release-3.0.0.md [hunk 2]
```diff
@@ -4,3 +4,166 @@
...
+
+**3. Positive Data Deletion**
+
+In distributed systems, common data deletion involves reverse GC, comparing files in shared storage with those recorded in metadata and deleting unrecorded files. This method can easily lead to data deletion errors under concurrent conditions. Relying on MetaService's transaction capabilities, Doris adopts asynchronous forward data deletion technology. Data generation is bound to a transaction, as is metadata deletion, and the asynchronous deletion process simply deletes data based on transaction records, eliminating the need for reverse GC.
+
+
+
+## Optimizer
+
+- All DQL and DML, except for load operations, will use the new optimizer. Default settings disable fallback to the old optimizer, which will be removed in subsequent versions.
```
Comment (johnpyp) [inline versioned_docs/version-3.0/releasenotes/release-3.0.0.md:105]: Suggestion: Elaborate on what the new optimizer is / link to docs to learn more.

File: versioned_docs/version-3.0/releasenotes/release-3.0.0.md [hunk 3]
```diff
@@ -4,3 +4,166 @@
...
+
+- Experimental Feature: Partition-level statistics. By activating the switch `set global enable_partition_analyze = true`, partition-level statistics can be collected and utilized, improving query performance in cases of data partition skew.
+
+- Experimental Feature: Generated columns. When creating tables, columns can be designated as generated columns, which automatically compute results based on defined expressions at write time, offering more complex expressions compared to default values but disallowing explicit writing of specified values.
+
+## Execution Engine
+
+- Experimental Feature: Spill capability for core operators such as join, aggregate, and sort has reached GA usable status.
+
+- Support for UDTFs.
```
Comment (johnpyp) [inline versioned_docs/version-3.0/releasenotes/release-3.0.0.md:119]: Suggestion: Link to information about UDTFs

File: versioned_docs/version-3.0/releasenotes/release-3.0.0.md [hunk 4]
```diff
@@ -4,3 +4,166 @@
...
+
+- Experimental Feature: Refactored logic for synchronized materialized view selection, transitioning from RBO to CBO to maintain consistency with asynchronous materialized views. If issues arise, the switch `set global enable_sync_mv_cost_based_rewrite = false` can be used to revert to RBO mode.
+
+- Experimental Feature: Partition-level statistics. By activating the switch `set global enable_partition_analyze = true`, partition-level statistics can be collected and utilized, improving query performance in cases of data partition skew.
+
+- Experimental Feature: Generated columns. When creating tables, columns can be designated as generated columns, which automatically compute results based on defined expressions at write time, offering more complex expressions compared to default values but disallowing explicit writing of specified values.
+
+## Execution Engine
+
+- Experimental Feature: Spill capability for core operators such as join, aggregate, and sort has reached GA usable status.
```
Comment (johnpyp) [inline versioned_docs/version-3.0/releasenotes/release-3.0.0.md:117]: It's unclear to the reader what this means... is it experimental? or GA? both?

I'd also link to the docs explaining.

File: versioned_docs/version-3.0/releasenotes/release-3.0.0.md [hunk 5]
```diff
@@ -4,3 +4,166 @@
...
+
+
+
+Based on these considerations, the architectural choices made include:
+
+1. A single metadata service divided into semantic and storage layers, with the storage layer choosing production-verified and transaction-capable FoundationDB.
+
+2. Storage options include mature, low-cost object storage and HDFS.
+
+3. To compensate for performance loss due to remote storage, local data caching has been introduced at computing nodes, combining the simplicity of a Shared Disk architecture with the performance of a Share Nothing architecture.
```
Comment (johnpyp) [inline versioned_docs/version-3.0/releasenotes/release-3.0.0.md:63]: If possible, information about the performance loss (with benchmarks) would be helpful.

**Review by johnpyp — COMMENTED** [review]
As a user, added some comments about things that would make this more readable/useful I think :)

**Pull request closed without merging**